src/gt4py/next/program_processors/runners/dace_fieldview/transformations/map_fusion_helper.py

```diff
@@ -165,22 +158,23 @@ def relocate_nodes(
         """
 
         # Now we relocate empty Memlets, from the `from_node` to the `to_node`
-        for empty_edge in filter(lambda e: e.data.is_empty(), state.out_edges(from_node)):
-            dace_helpers.redirect_edge(state, empty_edge, new_src=to_node)
-        for empty_edge in filter(lambda e: e.data.is_empty(), state.in_edges(from_node)):
-            dace_helpers.redirect_edge(state, empty_edge, new_dst=to_node)
+        for empty_edge in list(filter(lambda e: e.data.is_empty(), state.out_edges(from_node))):
```

Comment:
It is needed, because redirect edge will essentially remove and recreate an edge.
It is similar if you iterate over a `dict` and then modify that `dict`.

Reply:
I see, it makes sense now.